mir-ci/mir_ci/fixtures/servers.py

```diff
@@ -22,7 +22,8 @@ class ServerCap(Flag):
     DISPLAY_CONFIG = auto()
     SCREENCOPY = auto()
     INPUT_METHOD = auto()
-    ALL = FLOATING_WINDOWS | DRAG_AND_DROP | DISPLAY_CONFIG | SCREENCOPY | INPUT_METHOD
+    MIR_FLUTTER_APP = auto()
+    ALL = FLOATING_WINDOWS | DRAG_AND_DROP | DISPLAY_CONFIG | SCREENCOPY | INPUT_METHOD | MIR_FLUTTER_APP
```

Comment:
It's kinda weird that the rest of these are server capabilities, and then there's `MIR_FLUTTER_APP`. Could this instead be a `MIR_SHELL_PROTO` capability (denoting the support for the `mir_shell` protocol), and then have the test case run the flutter app against the server selected?

That would fit in better with how the rest of these tests work, and also allow us to test different servers that support `mir_shell`? (ie: maybe we want to test both miriway *and* miral-shell?)

Reply:
Yes, `mir_demo_server.mir_flutter_app` is just the demo server running an app. It doesn't seem to fit here, but do I have another option? Using `MIR_SHELL_PROTO` would work great if, instead of bundling `mir_flutter_app` into `mir_demo_server`, `mir_flutter_app` had its own snap. This was actually what I did in my [first commit](https://github.com/canonical/mir-ci/pull/118/commits/619ed41d42f2058ab71e416340617defb0a177de#diff-11d0b7d197b438cc57d573b5fa3f5ca9ba7fc6e40589f04661db1774b1a1c88b), running a [mir_flutter_app snap](https://github.com/hbatagelo/mir_flutter_app/blob/main/snap/snapcraft.yaml) in the servers that support mir_shell, like miriway and miral-shell (IIRC, I used MIR_SHELL to tag the compatible servers, but then changed to MIR_FLUTTER_APP).

In the current state, `MIR_SHELL_PROTO` can only be used with `mir_demo_server.mir_flutter_app` because  `test_mir_flutter_app.py` won't run any application as a subprocess. It we tag miriway or confined_server with `MIR_SHELL_PROTO`, the test will just time out because there will be no application to test.

Reply:
@hbatagelo why do we have the reference app "bundled" with a server? Why not just run a server like we usually do, and run the app itself separately, like we do in other tests?

Reply:
> @hbatagelo why do we have the reference app "bundled" with a server? Why not just run a server like we usually do, and run the app itself separately, like we do in other tests?

Oh, I thought MIRENG-479 was about that. I used the other mir-test-tools apps (sdl22-test, qt-test, etc) as a reference, as they are all invoked by mir_demo_server. I'll change the mir-flutter-app snap "app" so it calls mir_flutter_app directly.